build_projects/shared-build-targets-utils/Utils/PublishMutationUtilties.cs

```diff
@@ -22,7 +22,7 @@ public class PublishMutationUtilties
             foreach (var binaryName in binariesToBeDeleted)
             {
                 File.Delete(Path.Combine(path, $"{binaryName}{Constants.ExeSuffix}"));
-                File.Delete(Path.Combine(path, $"{binaryName}.dll"));
+                File.Delete(Path.Combine(path, $"{Constants.DynamicLibPrefix}{binaryName}{Constants.DynamicLibSuffix}"));
```

Comment:
> used to be used for managed assemblies

When this code was combined with CLI.

Ex https://github.com/dotnet/cli/blob/rel/1.0.0-preview2.1/build_projects/dotnet-cli-build/CompileTargets.cs#L236-L241

I agree it is no longer used for managed assemblies, and your change is safe.